src/webgpu/api/validation/image_copy/buffer_texture_copies.spec.ts

```diff
@@ -405,3 +405,51 @@ g.test('texture_buffer_usages')
       t.testCopyTextureToBuffer({ texture }, { buffer }, textureSize, isSuccess);
     }
   });
+
+g.test('device_mismatch')
+  .desc(
+    `
+    Tests copyBufferToTexture and copyTextureToBuffer cannot be called with a buffer or a texture
+    created from another device.
+  `
+  )
+  .params(u =>
+    u //
+      .combine('copyType', ['CopyB2T', 'CopyT2B'] as const)
+      .beginSubcases()
+      .combineWithParams([
+        { srcMismatched: false, dstMismatched: false }, // control case
+        { srcMismatched: true, dstMismatched: false },
+        { srcMismatched: false, dstMismatched: true },
+      ] as const)
+  )
+  .beforeAllSubcases(t => {
+    t.selectMismatchedDeviceOrSkipTestCase(undefined);
+  })
+  .fn(async t => {
+    const { copyType, srcMismatched, dstMismatched } = t.params;
+    const mismatched = srcMismatched || dstMismatched;
+
+    const device = mismatched ? t.mismatchedDevice : t.device;
```

Comment:
This selects the device for both the buffer and the texture. I think what you want is more like:

`(bufMismatched ? t.mismatchedDevice : t.device).createBuffer`

`(texMismatched ? t.mismatchedDevice : t.device).createTexture`

The variable `mismatched` should be renamed to something like `invalid` and moved down to near the bottom where it's used.

Reply:
Thank you for pointing it out. Updated.